### PR TITLE
Do not error for empty get nodegroups if output is yaml or json

### PR DESCRIPTION
**Pull request by Callisto13**
### Description

Fixes https://github.com/weaveworks/eksctl/issues/3177

When running `get nodegroup` returns no nodegroups, we should only report that as an error if the output is `table`. For `yaml` and `json` we should return an empty object.

```sh
$ ./eksctl get nodegroup --cluster cb-test2 -o table
[ℹ]  eksctl version 0.38.0-dev+d09881f8.2021-01-29T12:27:01Z
[ℹ]  using region us-west-2
Error: No nodegroups found

$ ./eksctl get nodegroup --cluster cb-test2 -o json
[]

$ ./eksctl get nodegroup --cluster cb-test2 -o yaml
[]
```

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/ctl/get/nodegroup.go [hunk 1]
```diff
@@ -90,20 +90,21 @@ func doGetNodeGroup(cmd *cmdutils.Cmd, ng *api.NodeGroup, params *getCmdParams)
...
 
 	if params.output == "table" {
+		// Empty summary implies no nodegroups
+		// We only error if the output is table, since if the output
+		// is yaml or json we should return an empty object.
+		if len(summaries) == 0 {
+			if ng.Name == "" {
+				return errors.Errorf("No nodegroups found")
+			}
+			return errors.Errorf("Nodegroup with name %v not found", ng.Name)
```
Comment (cPu1) [inline pkg/ctl/get/nodegroup.go:106]: Super-minor nit: 
```suggestion
			return errors.Errorf("nodegroup with name %v not found", ng.Name)
```

(I know this is taken from the surrounding block)

**Review by cPu1 — APPROVED** [review]

**Review by michaelbeaumont — APPROVED** [review]
Hmm we just had https://github.com/weaveworks/eksctl/pull/2873 too

**Comment by Callisto13** [comment]
hmmm interesting

**Comment by Callisto13** [comment]
Screenshot 2021-01-29 at 12 58 55
awwww 😞

**Pull request merged**